xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-api/src/main/java/org/xwiki/security/authorization/Right.java

```diff
@@ -373,7 +413,10 @@ public String toString()
     @Override
     public Set<Right> getImpliedRights()
     {
-        return impliedRights;
+        if (this.immutableImpliedRights == null && !impliedRights.isEmpty()) {
```

Comment:
The if should only have `this.immutableImpliedRights == null`. If the set is empty anyone will be able to add new rights in it through the Set returned by getImpliedRights() returns.

Reply:
>  If the set is empty anyone will be able to add new rights in it through the Set returned by getImpliedRights() returns.

I don't understand: the idea here is to keep returning null if `impliedRights` has always been empty.

Reply:
Right, I misread. For some reason in my mind it meant we would return impliedRights directly if it was empty but immutableImpliedRights is always null when impliedRights is empty.